modules/nf-core/matrixfilter/meta.yml

```diff
@@ -0,0 +1,71 @@
+name: "matrixfilter"
+description: filter a matrix based on a minimum value and numbers of samples
+  that must pass.
+keywords:
+  - matrix
+  - filter
+tools:
+  - "matrixfilter":
+      description: "filter a matrix based on a minimum value and numbers of samples"
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing information on experiment, at a minimum an id.
+        e.g. [ id:'test' ]
+  - samplesheet:
+      type: file
+      description: |
+        CSV or TSV format sample sheet with sample metadata
+  - abundance:
+      type: file
+      description: |
+        Raw TSV or CSV format expression abundance matrix with features (e.g.
+        genes) by row and observations (e.g. samples) by column. All rowsnames
+        from the sample sheet should be present in the columns.
+  - minimum_abundance:
+      type: numeric
+      description: |
+        Minimum abundance value, supplied via task.ext.args as --minimum_abundance
+      default: 1
+  - minimum_samples:
+      type: numeric
+      description: |
+        Minimum observations that must pass the threshold to retain
+        the row/ feature (e.g. gene). Supplied via task.ext.args as
+        --minimum_samples
+      default: 1
+  - minimum_proportion:
+      type: numeric
+      description: |
+        A minimum proportion of observations that must pass the threshold.
+        Supplied via task.ext.args as --minimum_proportion. Overrides
+        minimum_samples
+      default: 0
+  - grouping_variable:
+      type: string
```

Comment:
I'm not sure about these to be honest, as this meta file is meant to just describe channels (but I don't have a better alternative place for it yet).

Reply:
I agree. The need to document optional params because they're not documented as part of an underlying tool is one of the reasons I raised this as a tricky case.